chiselFrontend/src/main/scala/chisel3/aop/Aspect.scala

```diff
@@ -0,0 +1,45 @@
+// See LICENSE for license details.
+
+package chisel3.aop
+
+import chisel3.experimental.{ChiselAnnotation, RawModule}
+import firrtl.annotations.Annotation
+import firrtl.options.{HasShellOptions, OptionsException, ShellOption, Unserializable}
+import firrtl.{AnnotationSeq, RenameMap, Transform}
+
+import scala.reflect.runtime.universe.TypeTag
+
+/** Represents an aspect of a Chisel module, by specifying
+  *  what behavior should be done to instance, via the FIRRTL Annotation Mechanism
+  * @param dutTag Needed to prevent type-erasure of the top-level module type
+  * @tparam T Type of top-level module
+  */
+abstract class Aspect[T <: RawModule](implicit dutTag: TypeTag[T]) extends Annotation with Unserializable {
```

Comment:
Is the `TypeTag` necessary here?

Reply:
Nope, it isn't, I'll remove it. The only thing is its usually pretty important for Aspect descendants to have a typetag.